File: jobs/2026-04-23__10-43-59/b2b_org_settings__D5LNR9E/artifacts/update_org.js
const stytch = require('stytch');

const projectId = process.env.STYTCH_PROJECT_ID;
const secret = process.env.STYTCH_SECRET;
const organization_id = process.argv[2];

if (!projectId || !secret) {
  console.error("Missing STYTCH_PROJECT_ID or STYTCH_SECRET");
  process.exit(1);
}

if (!organization_id) {
  console.error("Please provide an organization_id");
  process.exit(1);
}

const client = new stytch.B2BClient({
  project_id: projectId,
  secret: secret,
});

async function updateOrg() {
  try {
    await client.organizations.update({
      organization_id: organization_id,
      auth_methods: 'RESTRICTED',
      allowed_auth_methods: ['sso', 'magic_link'],
      email_jit_provisioning: 'RESTRICTED'
    });
    console.log("Organization updated successfully");
  } catch (error) {
    console.error("Error updating organization:", error);
    process.exit(1);
  }
}

updateOrg();

File: jobs/2026-04-26__00-13-54/b2b_update_organization__uysNoyy/artifacts/update_org.py
#!/usr/bin/env python3
import os
import sys

import stytch


def get_env_var(name: str) -> str:
    value = os.getenv(name)
    if not value:
        raise RuntimeError(f"Missing required environment variable: {name}")
    return value


def read_trial_id(path: str) -> str:
    try:
        with open(path, "r", encoding="utf-8") as handle:
            return handle.read().strip()
    except FileNotFoundError as exc:
        raise RuntimeError(f"trial_id file not found at {path}") from exc


def extract_org_id(response) -> str:
    org_id = getattr(response, "organization_id", None)
    if org_id:
        return org_id

    org = getattr(response, "organization", None)
    if org:
        org_id = getattr(org, "organization_id", None)
        if org_id:
            return org_id

    raise RuntimeError("Unable to determine organization_id from create response")


def main() -> int:
    project_id = get_env_var("STYTCH_B2B_PROJECT_ID")
    secret = get_env_var("STYTCH_B2B_SECRET")
    trial_id = read_trial_id("/logs/trial_id")

    client = stytch.B2BClient(project_id=project_id, secret=secret)

    organization_name = f"test-org-{trial_id}"
    organization_slug = f"test-org-{trial_id}"

    create_response = client.organizations.create(
        organization_name=organization_name,
        organization_slug=organization_slug,
    )

    organization_id = extract_org_id(create_response)

    client.organizations.update(
        organization_id=organization_id,
        auth_methods="RESTRICTED",
        allowed_auth_methods=["sso", "magic_link"],
    )

    print(
        "Organization created and updated successfully:",
        f"id={organization_id}",
        f"name={organization_name}",
    )
    return 0


if __name__ == "__main__":
    try:
        sys.exit(main())
    except Exception as exc:  # pragma: no cover - surface all errors
        print(f"Error: {exc}", file=sys.stderr)
        sys.exit(1)
